src/main/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataHelper.java

```diff
@@ -0,0 +1,356 @@
+import org.opensearch.OpenSearchException;
+import org.opensearch.SpecialPermission;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.action.bulk.BulkRequest;
+import org.opensearch.action.bulk.BulkResponse;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.search.MultiSearchRequestBuilder;
+import org.opensearch.action.search.MultiSearchResponse;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.client.Client;
+import org.opensearch.client.Requests;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.SuppressForbidden;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.index.query.QueryBuilders;
+
+/**
+ * Helper class for GeoIp data
+ */
+@Log4j2
+public class GeoIpDataHelper {
+    private static final String IP_RANGE_FIELD_NAME = "_cidr";
+    private static final String DATA_FIELD_NAME = "_data";
+
+    /**
+     * Create an index of single shard with auto expand replicas to all nodes
+     *
+     * @param clusterService cluster service
+     * @param client client
+     * @param indexName index name
+     * @param timeout timeout
+     * @throws IOException io exception
+     */
+    public static void createIndex(
+        final ClusterService clusterService,
+        final Client client,
+        final String indexName,
+        final TimeValue timeout
+    ) throws IOException {
+        if (clusterService.state().metadata().hasIndex(indexName) == true) {
+            log.info("Index {} already exist. Skipping creation.", indexName);
+            return;
+        }
+        final Map<String, Object> indexSettings = new HashMap<>();
+        indexSettings.put("index.number_of_shards", 1);
+        indexSettings.put("index.auto_expand_replicas", "0-all");
+        final CreateIndexRequest createIndexRequest = new CreateIndexRequest(indexName).settings(indexSettings).mapping(getIndexMapping());
+        CreateIndexResponse response = client.admin().indices().create(createIndexRequest).actionGet(timeout);
+    }
+
+    /**
+     * Generate XContentBuilder representing datasource database index mapping
+     *
+     * {
+     *     "dynamic": false,
+     *     "properties": {
+     *         "_cidr": {
+     *             "type": "ip_range",
+     *             "doc_values": false
+     *         }
+     *     }
+     * }
+     *
+     * @return String representing datasource database index mapping
+     */
+    private static String getIndexMapping() {
+        try {
+            try (InputStream is = DatasourceHelper.class.getResourceAsStream("/mappings/ip2geo_datasource.json")) {
```

Comment:
what is advantage of storing default mappings in file over variable inside class?

Reply:
Easier to read/write mapping in text file than having them in java file for human.

Reply:
Then you have to deserialize that file every time , which will cost you some latency. Since your mapping file contains few fields, i don't think we need a separate file.

Reply:
1. It does not get call very frequently. 
2. It does not impact latency as it is called in background task but not as part of API call.